src/state/selectors/barSelectors.ts

```diff
@@ -0,0 +1,495 @@
+    return selectStackGroups(state, 'yAxis', yAxisId);
+  }
+  return selectStackGroups(state, 'xAxis', xAxisId);
+};
+
+export const selectBarCartesianAxisSize = (state: RechartsRootState, xAxisId: AxisId, yAxisId: AxisId) => {
+  const layout = selectChartLayout(state);
+  if (layout === 'horizontal') {
+    return selectCartesianAxisSize(state, 'xAxis', xAxisId);
+  }
+  return selectCartesianAxisSize(state, 'yAxis', yAxisId);
+};
+
+export const selectBarSizeList: (
+  state: RechartsRootState,
+  xAxisId: AxisId,
+  yAxisId: AxisId,
+  isPanorama: boolean,
+  barSettings: BarSettings,
+) => SizeList | undefined = createSelector(
+  [selectAllVisibleBars, selectRootBarSize, selectBarCartesianAxisSize],
+  (allBars: ReadonlyArray<CartesianGraphicalItemSettings>, globalSize: number | undefined, totalSize) => {
+    const initialValue: Record<StackId, Array<CartesianGraphicalItemSettings>> = {};
+
+    const stackedBars = allBars.filter(b => b.stackId != null);
+    const unstackedBars = allBars.filter(b => b.stackId == null);
+
+    const groupByStack: Record<StackId, Array<CartesianGraphicalItemSettings>> = stackedBars.reduce((acc, bar) => {
+      if (!acc[bar.stackId]) {
+        acc[bar.stackId] = [];
+      }
+      acc[bar.stackId].push(bar);
+      return acc;
+    }, initialValue);
+
+    const stackedSizeList = Object.entries(groupByStack).map(([stackId, bars]) => {
+      const dataKeys = bars.map(b => b.dataKey);
+      const barSize = getBarSize(globalSize, totalSize, bars[0].barSize);
+      return { stackId, dataKeys, barSize };
+    });
+
+    const unstackedSizeList = unstackedBars.map(b => {
+      const dataKeys = [b.dataKey];
+      const barSize = getBarSize(globalSize, totalSize, b.barSize);
+      return { stackId: undefined, dataKeys, barSize };
+    });
+
+    return [...stackedSizeList, ...unstackedSizeList];
+  },
+);
+
+export const selectBarBandSize: (
+  state: RechartsRootState,
+  xAxisId: AxisId,
+  yAxisId: AxisId,
+  isPanorama: boolean,
+  barSettings: BarSettings,
+) => number | undefined = (
+  state: RechartsRootState,
+  xAxisId: AxisId,
+  yAxisId: AxisId,
+  isPanorama: boolean,
+  barSettings: BarSettings,
+): number | undefined => {
+  const layout = selectChartLayout(state);
+  const globalMaxBarSize = selectRootMaxBarSize(state);
+  const { maxBarSize: childMaxBarSize } = barSettings;
+  const maxBarSize: number = isNil(childMaxBarSize) ? globalMaxBarSize : childMaxBarSize;
+  let axis: BaseAxisWithScale, ticks: ReadonlyArray<TickItem>;
+  if (layout === 'horizontal') {
+    axis = selectAxisWithScale(state, 'xAxis', xAxisId, isPanorama);
+    ticks = selectTicksOfGraphicalItem(state, 'xAxis', xAxisId, isPanorama);
+  } else {
+    axis = selectAxisWithScale(state, 'yAxis', yAxisId, isPanorama);
+    ticks = selectTicksOfGraphicalItem(state, 'yAxis', yAxisId, isPanorama);
+  }
+  return getBandSizeOfAxis(axis, ticks, true) ?? maxBarSize ?? 0;
+};
+
+const selectAxisBandSize = (
```

Comment:
whats the difference between bar band size and axis band size?

Reply:
I don't understand this well enough to be able to explain. I reproduced what was in the generator.